src/components/Map/Map.jsx

```diff
@@ -1,41 +1,73 @@
-'use client'
-import {MapContainer, Marker, Popup, TileLayer} from "react-leaflet";
+"use client";
+
+// IMPORTANT: the order matters!
 import "leaflet/dist/leaflet.css";
-import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.webpack.css"; // Re-uses images from ~leaflet package
+import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.webpack.css";
 import "leaflet-defaulticon-compatibility";
-import {ItemMarker} from "@/components/Map/ItemMarker";
+
+import { useRef, useEffect } from "react";
+import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
+import { SelectionOverlay } from "@/components/Map/SelectionOverlay";
 import L from "leaflet";
-import {SelectionOverlay} from "@/components/Map/SelectionOverlay";
-import {useChatMutation} from "@/store/services/chatApi";
 
-export const Map = (props) => {
-  const {position} = props;
-  const icon = L.icon({iconUrl: "/images/marker-icon.png"});
-  const [chat, { isLoading, isSuccess, data, error }] = useChatMutation();
+const Map = (props) => {
+  const { position, placesData: places } = props;
+
+  const mapRef = useRef();
+
+  useEffect(() => {
+    // Wait until the map is initialized
+    if (!mapRef.current) return;
+
+    console.log("Running effect", places);
+    // Add new markers for the data received
+    const markers = places.map((item) => {
```

Comment:
Please, use ItemMarker component to render the markers

Reply:
Used ItemMarker, but also stripped ItemMarker quite a lot, as I don't understand why there was so much logic that is already implemented with React leaflet. 
If we need some functionality inside the popup, we need to check into react-leaflet-enhanced-marker package.

Reply:
We have to have both tooltip for showing the location name and Popup with extra infromation. The logic that you removed was for managing state of the marker and tooltip (opened / closed) from React, Otherwise it's uncontrolled element that behave the way leaflet developers wanted it to behave. I don't get what exactly the issue with this component so that it should be removed / replaced?

Reply:
If I try to render them within the component, like:

```
        {places &&
          places.map((item) => {
            return <ItemMarker item={item} />;
          })}
```

I receive the following error once it tries to render the markers:

```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading '_leaflet_events')

Call Stack
removeOne
node_modules/leaflet/dist/leaflet-src.js (2839:1)
off
node_modules/leaflet/dist/leaflet-src.js (2770:1)
NewClass._removeIcon
node_modules/leaflet/dist/leaflet-src.js (7970:1)
NewClass.onRemove
node_modules/leaflet/dist/leaflet-src.js (7815:1)
NewClass.removeLayer
node_modules/leaflet/dist/leaflet-src.js (6968:1)
removeLayer
node_modules/@react-leaflet/core/lib/layer.js (12:1)

```

And looking at the react-leaflet documentation, I just implemented it how their API said.
Tooltip is there (marker.addTooltip), and the exactly same information is provided in the popup, that's why I said that if we need some more advanced logic inside the popup, we should consider implementing our own.

If you are able to get it to work easily by removing the useEffect and rendering them inside the MapContainer, feel free to commit.

Reply:
I'll take a look later today, thank for giving more context

Reply:
I will make a commit so that the ItemMarker component is not changed, so this can be merged.

Reply:
Pushed a commit to reset the ItemMarker component, also created a trello for this: 
https://trello.com/c/G1I1Wjih/30-use-itemmarker-component-when-rendering-markers-on-the-map

Please approve and we could merge this and work on top of this.